staging/src/k8s.io/client-go/applyconfigurations/authorization/v1/subjectaccessreviewspec.go

```diff
@@ -0,0 +1,76 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by applyconfiguration-gen. DO NOT EDIT.
+
+package v1
+
+import (
+	authorizationv1 "k8s.io/api/authorization/v1"
+)
+
+// SubjectAccessReviewSpecApplyConfiguration represents an declarative configuration of the SubjectAccessReviewSpec type for use
+// with apply.
+type SubjectAccessReviewSpecApplyConfiguration struct {
+	ResourceAttributes    *ResourceAttributesApplyConfiguration    `json:"resourceAttributes,omitempty"`
+	NonResourceAttributes *NonResourceAttributesApplyConfiguration `json:"nonResourceAttributes,omitempty"`
+	User                  *string                                  `json:"user,omitempty"`
+	Groups                []string                                 `json:"groups,omitempty"`
+	Extra                 map[string]authorizationv1.ExtraValue    `json:"extra,omitempty"`
+	UID                   *string                                  `json:"uid,omitempty"`
+}
+
+// SubjectAccessReviewSpecApplyConfiguration constructs an declarative configuration of the SubjectAccessReviewSpec type for use with
+// apply.
+func SubjectAccessReviewSpec() *SubjectAccessReviewSpecApplyConfiguration {
+	return &SubjectAccessReviewSpecApplyConfiguration{}
+}
+
+// SetResourceAttributes sets the ResourceAttributes field in the declarative configuration to the given value.
+func (b *SubjectAccessReviewSpecApplyConfiguration) SetResourceAttributes(value *ResourceAttributesApplyConfiguration) *SubjectAccessReviewSpecApplyConfiguration {
+	b.ResourceAttributes = value
+	return b
+}
+
+// SetNonResourceAttributes sets the NonResourceAttributes field in the declarative configuration to the given value.
+func (b *SubjectAccessReviewSpecApplyConfiguration) SetNonResourceAttributes(value *NonResourceAttributesApplyConfiguration) *SubjectAccessReviewSpecApplyConfiguration {
+	b.NonResourceAttributes = value
+	return b
+}
+
+// SetUser sets the User field in the declarative configuration to the given value.
+func (b *SubjectAccessReviewSpecApplyConfiguration) SetUser(value string) *SubjectAccessReviewSpecApplyConfiguration {
+	b.User = &value
+	return b
+}
+
+// SetGroups sets the Groups field in the declarative configuration to the given value.
+func (b *SubjectAccessReviewSpecApplyConfiguration) SetGroups(value []string) *SubjectAccessReviewSpecApplyConfiguration {
+	b.Groups = value
+	return b
+}
+
+// SetExtra sets the Extra field in the declarative configuration to the given value.
+func (b *SubjectAccessReviewSpecApplyConfiguration) SetExtra(value map[string]authorizationv1.ExtraValue) *SubjectAccessReviewSpecApplyConfiguration {
```

Comment:
If we do make list take pointers, we should be consistent for map, I think. In case someone wants to set a null item in the map?

Reply:
What's going on here might be a bit confusing.

We're only making lists of apply configuration types take pointers. No maps of application configuration types exist in k/k (because people have been following the rule of making associative lists instead), but if they did, we would generate a pointer.

`ExtraValue` is a typeref to `[]string` and we don't add a pointer for it (for maps or lists).